tests/functional/aws-node-sdk/test/object/rangeTest.js

```diff
@@ -0,0 +1,254 @@
+import assert from 'assert';
+import async from 'async';
+import { exec, execFile } from 'child_process';
+import fs from 'fs';
+
+import withV4 from '../support/withV4';
+import BucketUtility from '../../lib/utility/bucket-util';
+
+const bucket = 'bucket-for-range-test';
+const key = 'key-for-range-test';
+let s3;
+
+// Calculate the total number of bytes for range end values
+const getTotal = (begin, end) =>
+    Number.parseInt(end, 10) - Number.parseInt(begin, 10) + 1;
+
+// Get the expected end values for various ranges (e.g., '-10', '10-', '-')
+const getOuterRange = (range, fileSize) => {
+    const arr = range.split('-');
+    if (arr[0] === '' && arr[1] !== '') {
+        arr[0] = Number.parseInt(fileSize, 10) - Number.parseInt(arr[1], 10);
+        arr[1] = Number.parseInt(fileSize, 10) - 1;
+    } else {
+        arr[0] = arr[0] === '' ? '0' : arr[0]; // For invalid ranges
+        arr[1] = arr[1] === '' || Number.parseInt(arr[1], 10) >= fileSize ?
+            Number.parseInt(fileSize, 10) - 1 : arr[1];
+    }
+    return {
+        begin: arr[0],
+        end: arr[1],
+    };
+};
+
+// Compare the S3 object from getObject with the corresponding hashed range file
+const checkRanges = (range, fileSize, cb) => {
+    s3.getObjectAsync({
+        Bucket: bucket,
+        Key: key,
+        Range: `bytes=${range}`,
+    })
+    .then(res => fs.readFile(`hashedFile.${fileSize}.${range}`, (err, data) => {
+        if (err) {
+            return cb(err);
+        }
+        const { begin, end } = getOuterRange(range, fileSize);
+        const contentLength = getTotal(begin, end).toString();
+        // If the range header is invalid, content range should be `undefined`
+        const contentRange = range === '-' ? undefined :
+            `bytes ${begin}-${end}/${fileSize}`;
+
+        assert.deepStrictEqual(res.ContentLength, contentLength);
+        assert.deepStrictEqual(res.ContentRange, contentRange);
+        assert.deepStrictEqual(res.ContentType, 'application/octet-stream');
+        assert.deepStrictEqual(res.Metadata, {});
+        assert.deepStrictEqual(res.Body, data);
+        return cb();
+    }))
+    .catch(cb);
+};
+
+// Complete the MPU, or abort it if there is an error
+const completeMPU = (range, fileSize, uploadId, cb) =>
+    s3.completeMultipartUploadAsync({
+        Bucket: bucket,
+        Key: key,
+        UploadId: uploadId,
+        MultipartUpload: {
+            Parts: [
+                {
+                    ETag: 'fa5a73131c61ee90e128fe6bef8544e2',
+                    PartNumber: 1,
+                },
+                {
+                    ETag: 'e09d29b0c1da7495508d56a4047fec71',
+                    PartNumber: 2,
+                },
+            ],
+        },
+    })
+    .then(() => checkRanges(range, fileSize, cb))
```

Comment:
With the refactoring of `beforeEach` and `afterEach`, this is now self contained.